web_invalid_tab/static/src/js/view_form.js

```diff
@@ -0,0 +1,62 @@
+openerp.web_invalid_tab = function(instance) {
+
+    var tab_selector = 'div[role="tabpanel"]';
+
+    function tab_link(tab) {
+        return $("a[href='#" + tab.attr('id') + "']");
+    }
+
+    function is_visible(tab, element) {
+        if ($(element).hasClass('oe_form_invisible') || element.style.display == 'none') {
```

Comment:
I'm not sure to understand well what you want.

Let's say there is field well visible (and empty) in the current tab (tab 1).
I then switch to another tab (tab 2).
This field will not change, and will not be visible anymore. it's tab 2 which will have his property display set to none.
is:visible gives you this info without the need to traverse the DOM and query each layer